src/execution_plan/ops/op_load_csv.c

```
@@ -17,23 +18,131 @@
+	if(pipe(pipefd) == -1) {
+		ErrorCtx_RaiseRuntimeException("Error creating pipe");
+		return NULL;
+	}
+
+	// download remote file
+	// file content will be written to pipe
+	FILE *f = fdopen(pipefd[1], "wb");
+	op->curl = Curl_Download(uri, &f);
+	if(op->curl == NULL) {
+		// close pipe read end, write-end closed by Curl_Download
+		close(pipefd[0]);
```

cppcheck (2.10-2)

[error] Memory leak

(memleak)